pyblazing/pyblazing/apiv2/context.py

```diff
@@ -201,22 +206,35 @@ def collectPartitionsRunQuery(
                     tables[table_name].input.append(worker.data[key])
 
     try:
-        result = cio.runQueryCaller(
+        dfs = cio.runQueryCaller(
                             masterIndex,
                             nodes,
                             tables,
                             fileTypes,
                             ctxToken,
                             algebra,
                             accessToken,
-                            config_options)
+                            config_options,
+                            is_single_node=False)
     except cio.RunQueryError as e:
         print(">>>>>>>> ", e)
-        result = cudf.DataFrame()
+        result = [cudf.DataFrame()]
     except Exception as e:
         raise e
 
-    return result
+    meta = dask.dataframe.utils.make_meta(dfs[0])
+    query_partids = []
+
+    with worker._lock:
+        if not hasattr(worker, "query_parts"):
+            worker.query_parts = {}
+
+    for df in dfs:
+        query_partid = random.randint(0, 64000) # query_partid should be a unique identifier
```

Comment:
increase this random to the range of an int32 just in case. And while you are at it, increase the range of the ctxToken

Reply:
this doesnt seem to handle any single_gpu cases